### ISSUE TITLE
psycopg2.extras.NamedTupleCursor leaks connections?

### ISSUE THREAD

**Issue by alexpdp7** [psycopg/psycopg2#981]
If I define:

```
import os
import psycopg2
import psycopg2.extras

def get_dsn():
    return os.environ['DSN']

def get_connection():
    return psycopg2.connect(get_dsn())                                                                                                                                   
```

with DSN pointing to a Postgres database (10, in my tests), and I run:

```
for i in range(1, 200):
  with get_connection() as conn:
    with conn.cursor() as cur:
       cur.execute("select 1")
       cur.fetchall()
```

I get a steady stream of `[(1,)]`. But if I do:

```
for i in range(1, 200):
  with get_connection() as conn:
    with conn.cursor(cursor_factory=psycopg2.extras.NamedTupleCursor) as cur:
       cur.execute("select 1")
       cur.fetchall()
```

, it breaks eventually:

```
...
[Record(f_column_=1)]
[Record(f_column_=1)]
[Record(f_column_=1)]
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in get_connection
  File "/usr/local/lib/python3.6/dist-packages/psycopg2/__init__.py", line 126, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
```

**Comment by alexpdp7** [comment]
Note that this also fails if I do:

```
>>> for i in range(1, 200):
...   with get_connection() as conn:
...     with conn.cursor(cursor_factory=psycopg2.extras.NamedTupleCursor) as cur:
...        cur.execute("select 1")
...        cur.fetchall()
...   conn.close()
```

**Comment by dvarrazzo** [comment]
With what version of psycopg? Is it the same problem of #928 fixed in 2.8.3?

**Comment by alexpdp7** [comment]
Doh, I dumb, read that bug but didn't think it was related. Yeah, 2.8.3 fixes this, sorry.
